org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/ClassFileWorkingCopy.java

```diff
@@ -79,6 +83,26 @@ public IPath getPath() {
 	return this.classFile.getPath();
 }
 
+@Override
+public char[] getFileName(){
+	char[] ret = null;
+	PackageFragmentRoot root = getPackageFragmentRoot();
+	if (root == null)
+		// working copy not in workspace
+		ret = new Path(getElementName()).toString().toCharArray();
+	else if (root.isArchive()) {
+		String clazzFileName = this.classFile.getElementName();
+		String parentPath = this.classFile.getParent().getPath().toString();
+		IPackageFragment enclosingPackage = (IPackageFragment)getAncestor(IJavaElement.PACKAGE_FRAGMENT);
+		String pack = enclosingPackage == null ? "" : enclosingPackage.getElementName();
+		String packReplaced = pack.length() > 0 ? pack.replaceAll("\\.", "/") + "/" : "";
```

Comment:
where do you see a requirement to use a specific separator?

Reply:
The documentation for IDependent.getFileName(); 

In summary, the two examples provided are:
```
 *  "c:\\lib\\some.jar|/com/p/X.class" or
 *  "/lib/some.zip|/com/q/Y.class".
```

It uses a system path for the resource, followed by the jar separator, followed by the package with forward slashes only for both major operating systems.  

```
/**
 * Answer the file name which defines the type.
 *
 * The path part (optional) must be separated from the actual
 * file proper name by a separator suitable for the type (java.io.File.separator for example),
 * e.g.
 *  "c:\\source\\com\\p\\X.java" or
 *  "/com/p/Y.java".
 *
 * The path to the zip or jar file (optional) must be separated
 * from the actual path part by JAR_FILE_ENTRY_SEPARATOR,
 * e.g.
 *  "c:\\lib\\some.jar|/com/p/X.class" or
 *  "/lib/some.zip|/com/q/Y.class".
 *
 * The proper file name includes the suffix extension (e.g.&nbsp;".java")
 * e.g.&nbsp;"c:/org/eclipse/jdt/internal/compileri/env/IDependent.java"
 *
 * Return null if no file defines the type.
 */

char[] getFileName();
```